### ISSUE TITLE
pdbqt protein

### ISSUE THREAD

**Issue by Golovin-Andrey** [oddt/oddt#14]
Maciej , I tried to implement oddt to student teaching with ipython notebooks and faced with wired thing:
I loaded protein 
```
for ps in oddt.toolkit.readfile('pdb','/tmp/1lmp.pdb'):
    if  ps.molwt > 5000:
        prot = ps
prot.calccharges()
```
made vina object
```
result= av.autodock_vina(protein=prot,auto_ligand=mol,size=(20,20,20),center=(42.0,48.0,30.0),executable='/usr/bin/vina',
```
and did the run
```
result.dock([mol],prot)
```
with following result
```
CalledProcessError: Command '['/usr/bin/vina', '--receptor', '/tmp/autodock_vina_mXvaAf/protein.pdbqt', '--ligand', '/tmp/autodock_vina_mXvaAf/ligands_i_spA1/0_.pdbqt', '--out', '/tmp/autodock_vina_mXvaAf/ligands_i_spA1/0__out.pdbqt', '--center_x', '0.00202614', '--center_y', '0.000736733', '--center_z', '1.17657e-05', '--size_x', '20', '--size_y', '20', '--size_z', '20', '--cpu', '1', '--exhaustiveness', '8', '--num_modes', '9', '--energy_range', '3']' returned non-zero exit status 1
```
turned out that pdbqt contain ROOT line and this leads to vina error
```
REMARK                         _______ _______ _______ _____ _____    ______ ____
ROOT
ATOM      1  N   LYS A  1       40.347  23.372  17.250  0.00  0.00    -0.118 NA
```
Actually it's problem of babel and this makes all workbench unusable since prepare_receptor4.py from MGL is dead with new numpy (oldnumeric issue). From this point you module looks very attractive as one possible way of making fast docking in ipython notebooks. 
I may add some stupid code to remove this lines inside     def set_protein(self, protein).
I just worried about better way

Andrey

**Comment by mwojcikowski** [comment]
It's actually a bug of Openbabel. I assume that you use 2.3.1 or 2.3.2 which is included with most distros ot there. I strongly advice to upgrade to anything newer, and this bug will disappear. There are few options beside compiling: conda and some 3rd party repos. If you use conda check out my channel [http://anaconda.org/mwojcikowski/packages] which contains both OpenBabel and ODDT. If you use ubuntu there are few choices, but I recommend my PPA [https://launchpad.net/~mwojcikowski/+archive/ubuntu/openbabel].

Obviously we could add those few lines of code to filter the ROOT line, although there is no way of checking in python which version of pybel/openbabel is used (I'm not couting calling "obabel -V"). 

Last but not least, if you pass a file directory of protein in pdbqt format which you fixed manually to result= av.autodock_vina(...) it will be passed untouched to vina executable.
`result= av.autodock_vina(protein='protein_file.pdbqt',auto_ligand=mol,size=(20,20,20),executable='/usr/bin/vina')`

To sum up: I don't think we should fix this issue on ODDT's side using a "hack", while it's already fixed elsewhere. There are ways to bay-pass this problem.

Let me know what do you think.

PS. Note that if you use `auto_ligand` the `center` patameter is not used, since the box center is determined as a centroid of ligand's heavy atoms.

**Comment by Golovin-Andrey** [comment]
Thanks a lot Maciej 
Compiled from git babel version works fine. I would mention somewhere that older versions of babel will not work. It's important since from babel site link leads to version 2.3.1 or 2.3.2
I made small sketch about making docking having smile. 
While printing docking output  i needed use this construction:

```
res = dock_obj.dock([mol],prot)
print dock_obj.params
print res[0].data.keys()
for r in res:
    print "%8.3f%8.3f\n" %(float(r.data['vina_rmsd_ub']),float(r.data['vina_affinity']))
```
data to be expected float comes out with string. 
Meanwhile I will make some tutorial how to use certain packages in user space. In computer class I don't have admin rights to install packages on server. 
To be continued...
Cheers, Andrey

**Comment by Golovin-Andrey** [comment]
Meanwhile i tried to implement flex docking today and it's not trivia, as i got babel doesn't really deal with residues

**Comment by mwojcikowski** [comment]
Conda is ideal if you don't have root access. I use it on regular basis on our cluster. Install miniconda first and then do `conda install -c mwojcikowski oddt` (this will pull also OpenBabel).

Data stored in data methods are generic data properties, which type can differ. Therefore we cannot cast them to float, since it may contain text. It's also true for RDKit. I don't know we could find a better solution, although I might be wrong. 

Flexible residues might be great feature. I will think how to implement it (openbabel implements resudues quite well, although they would have to be splitted to separate molecule object. If you would like to contribute you can create Pull Request or contact me and we'll disscuss it off the Github (via mail preferably - maciek@wojcikowski.pl).

**Comment by Golovin-Andrey** [comment]
Hi Maciej!! 
Sure I will try to contribute, just give me some time to get in.
I just returned to notebook and faced with a problem:
lets create mol 
```
prot = oddt.toolkit.readfile('pdb','/tmp/1lmp.pdb').next()
mol = oddt.toolkit.readstring('smi', 'c1cccc(O)c1')

if not mol.OBMol.Has3D(): 
    mol.make3D(forcefield='mmff94', steps=150)
    mol.localopt()
    mol.removeh()
    mol.OBMol.AddPolarHydrogens()
```
ok, go to docking
```
dock_obj= oddt.docking.AutodockVina.autodock_vina(
    protein=prot,size=(20,20,20),center=(42.0,48.0,30.0),
    executable='/usr/bin/vina',autocleanup=False, num_modes=20)
res = dock_obj.dock([mol],prot)
```

and show it nglview 

```
w1 = show_ob_structure(mol)
mol2= mol.clone
mol2.coords=res[0].coords
w2 = show_ob_structure(mol2)
Box(children=(w1,w2))
```

and all bonds became 1A long (pymol as viewer gives the same)
[screenshot](http://vsb.fbb.msu.ru/share/golovin/oddt.png)
I do something wrong... :)
